src/SFA.DAS.LearnerDataMismatches.UnitTests/Domain/AcademicYearTests.cs

```diff
@@ -1,39 +1,57 @@
-using System;
 using FluentAssertions;
 using NUnit.Framework;
 using SFA.DAS.LearnerDataMismatches.Domain;
+using System;
 
 namespace SFA.DAS.LearnerDataMismatches.UnitTests.Domain
 {
     public class AcademicYearTests
     {
-        static object[] AcademicYearTestData = 
+        private static readonly object[] AcademicYearTestData =
         {
             new object[] {new DateTime(2019,8,1), 1920, 1819},
             new object[] {new DateTime(2019,7,31), 1819, 1718}
         };
 
-        [TestCaseSource("AcademicYearTestData")]
+        [TestCaseSource(nameof(AcademicYearTestData))]
         public void GenerateCorrectAcademicYears(DateTime today, int expectedCurrentYear, int expectedPreviousYear)
         {
             var model = new AcademicYear(today);
 
-            model.ShortRepresentation.Should().Be(expectedCurrentYear);
-            (model - 1).ShortRepresentation.Should().Be(expectedPreviousYear);
+            model.Should().Be((AcademicYear)expectedCurrentYear);
```

Comment:
Test method arguments can be of AcademicYear instead of int so no casting needed with in the method.

Reply:
Unfortunately that throws
`System.ArgumentException : Object of type 'System.Int32' cannot be converted to type 'SFA.DAS.LearnerDataMismatches.Domain.AcademicYear'.`

I think NUnit is using reflection and executes the test methods by calling `InvokeMethod`, which expects the arguments to be of the required type already.  There's no compiler-time casting available to it.